tables/TaQL/test/tExprNodeSetOpt.cc

```diff
@@ -0,0 +1,361 @@
+    AlwaysAssertExit (set.contains (id, 14.));
+    AlwaysAssertExit (! set.contains (id, 14.5));
+    AlwaysAssertExit (set.contains (id, 21.));
+    AlwaysAssertExit (set.contains (id, 21.5));
+    AlwaysAssertExit (set.contains (id, 22.));
+    AlwaysAssertExit (! set.contains (id, 22.5));
+    AlwaysAssertExit (set.find(22.5) == -1);
+    AlwaysAssertExit (set.find(13.5) == 6);
+  }
+  {
+    // Test open-closed intervals
+    TableExprNodeSetOptContSetOC<Double> set (TableExprNodeSet(), st, end);
+    set.show(cout, 2);
+    AlwaysAssertExit (! set.contains (id, 0.5));
+    AlwaysAssertExit (! set.contains (id, 1.));
+    AlwaysAssertExit (set.contains (id, 1.5));
+    AlwaysAssertExit (set.contains (id, 2.));
+    AlwaysAssertExit (! set.contains (id, 2.5));
+    AlwaysAssertExit (! set.contains (id, 13.));
+    AlwaysAssertExit (set.contains (id, 13.5));
+    AlwaysAssertExit (set.contains (id, 14.));
+    AlwaysAssertExit (! set.contains (id, 14.5));
+    AlwaysAssertExit (! set.contains (id, 21.));
+    AlwaysAssertExit (set.contains (id, 21.5));
+    AlwaysAssertExit (set.contains (id, 22.));
+    AlwaysAssertExit (! set.contains (id, 22.5));
+  }
+  {
+    // Test closed-open intervals
+    TableExprNodeSetOptContSetCO<Double> set (TableExprNodeSet(), st, end);
+    set.show(cout, 2);
+    AlwaysAssertExit (! set.contains (id, 0.5));
+    AlwaysAssertExit (set.contains (id, 1.));
+    AlwaysAssertExit (set.contains (id, 1.5));
+    AlwaysAssertExit (! set.contains (id, 2.));
+    AlwaysAssertExit (! set.contains (id, 2.5));
+    AlwaysAssertExit (set.contains (id, 13.));
+    AlwaysAssertExit (set.contains (id, 13.5));
+    AlwaysAssertExit (! set.contains (id, 14.));
+    AlwaysAssertExit (! set.contains (id, 14.5));
+    AlwaysAssertExit (set.contains (id, 21.));
+    AlwaysAssertExit (set.contains (id, 21.5));
+    AlwaysAssertExit (! set.contains (id, 22.));
+    AlwaysAssertExit (! set.contains (id, 22.5));
+  }
+  {
+    // Test open-open intervals
+    TableExprNodeSetOptContSetOO<Double> set (TableExprNodeSet(), st, end);
+    set.show(cout, 2);
+    AlwaysAssertExit (! set.contains (id, 0.5));
+    AlwaysAssertExit (! set.contains (id, 1.));
+    AlwaysAssertExit (set.contains (id, 1.5));
+    AlwaysAssertExit (! set.contains (id, 2.));
+    AlwaysAssertExit (! set.contains (id, 2.5));
+    AlwaysAssertExit (! set.contains (id, 13.));
+    AlwaysAssertExit (set.contains (id, 13.5));
+    AlwaysAssertExit (! set.contains (id, 14.));
+    AlwaysAssertExit (! set.contains (id, 14.5));
+    AlwaysAssertExit (! set.contains (id, 21.));
+    AlwaysAssertExit (set.contains (id, 21.5));
+    AlwaysAssertExit (! set.contains (id, 22.));
+    AlwaysAssertExit (! set.contains (id, 22.5));
+  }
+  {
+    // Test intervals with mix of open and closed
+    std::vector<bool> leftC ({false,true,true,true,true,true,false,true,true,true,true});
+    std::vector<bool> rightC({false,true,true,true,true,true,true ,true,true,true,false});
```

Comment:
I really would like to have test with less elements. The input data is quite far from this part of the input and with all true. false mixes here it's hard to verify the results in the test are correct.